server/channels/app/limits.go

```diff
@@ -17,36 +15,37 @@ const (
 )
 
 func (a *App) GetServerLimits() (*model.ServerLimits, *model.AppError) {
-	var limits = &model.ServerLimits{}
-
-	if a.shouldShowUserLimits() {
-		activeUserCount, appErr := a.Srv().Store().User().Count(model.UserCountOptions{})
-		if appErr != nil {
-			mlog.Error("Failed to get active user count from database", mlog.String("error", appErr.Error()))
-			return nil, model.NewAppError("GetServerLimits", "app.limits.get_app_limits.user_count.store_error", nil, "", http.StatusInternalServerError).Wrap(appErr)
-		}
+	limits := &model.ServerLimits{}
+	license := a.License()
 
-		limits.ActiveUserCount = activeUserCount
+	if license == nil && maxUsersLimit > 0 {
+		// Enforce hard-coded limits for unlicensed servers.
 		limits.MaxUsersLimit = maxUsersLimit
 		limits.MaxUsersHardLimit = maxUsersHardLimit
+	} else if license.IsSeatCountEnforced && license.Features != nil && license.Features.Users != nil {
+		// Enforce license limits as required by the license.
+		limits.MaxUsersLimit = int64(*license.Features.Users)
+		limits.MaxUsersHardLimit = int64(*license.Features.Users)
```

Comment:
Reusing these exact same variables will trigger a banner in the webapp that refers to ERROR_SAFETY_LIMITS_EXCEEDED and links to https://mattermost.com/pl/error-code-error-safety-limits-exceeded which doesn't apply for licensed servers. So we need to do a change in that banner to not show if licensed

Reply:
@marianunez, sounds good -- I'll look out for that once I can test end-to-end.

Reply:
But you do see the banner of reached the seat limit correct?